Listener/EventListener.php

```diff
@@ -54,10 +54,11 @@ public function handleEvent($event, string $eventName): void
 
     public function onKernelResponse(ResponseEvent $event): void
     {
-        //We only need the master request in order to keep all the original request headers
-        //This will be used to resolve advanced configuration tags.
+        // We only need the master request in order to keep all the original request headers
+        // This will be used to resolve advanced configuration tags.
         // such as '@=request.get('queryParam')'
-        if ($event->isMasterRequest()) {
+        $isMainRequestMethod = method_exists($event, 'isMainRequest') ? 'isMainRequest' : 'isMasterRequest';
+        if ($event->$isMainRequestMethod()) {
```

Comment:
```suggestion
        $isMainRequest = method_exists($event, 'isMainRequest') ? $event->isMainRequest() : $event->isMasterRequest();
        if ($isMainRequest) {
```

to avoid using the dynamic function call feature of PHP (which is probably not the best feature out there... and is hard to validate when using static analysis tools)

Reply:
I think the easiest way is to wrap the `$event->isMasterRequest()` in a condition that phpstan can rely on:


```suggestion
        $isMainRequest = method_exists($event, 'isMainRequest') ? $event->isMainRequest() : (method_exists($event, 'isMasterRequest') ? $event->isMasterRequest() : false);
        if ($isMainRequest) {
```

it's ugly but it's not a performance issue to have 2 checks since the first check should always be `true` (in the future at least)

Reply:
since it complains anyway (method_exists($event, 'isMasterRequest') will always evaluate to false.) I kept your first version and ignored errors in phpstan conf.